pkg/webhook/certs.go

```
@@ -0,0 +1,486 @@
+	if err := c.Watch(&source.Kind{Type: vwh}, mapper); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileVWH{}
+
+// ReconcileVWH reconciles a validatingwebhookconfiguration, making sure it
+// has the appropriate CA cert
+type ReconcileVWH struct {
+	client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a validatingwebhookconfiguration
+// object and makes sure the most recent CA cert is included
+func (r *ReconcileVWH) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.NamespacedName != secretKey {
+		return reconcile.Result{}, nil
+	}
+	secret := &corev1.Secret{}
+	if err := r.Get(context.TODO(), request.NamespacedName, secret); err != nil {
+		if k8sErrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	vwh := &unstructured.Unstructured{}
+	vwh.SetGroupVersionKind(vwhGVK)
+	if err := r.Get(context.TODO(), vwhKey, vwh); err != nil {
+		if k8sErrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	if secret.GetDeletionTimestamp().IsZero() {
+		artifacts, err := buildArtifactsFromSecret(secret)
+		if err != nil {
+			log.Error(err, "secret is not well-formed, cannot update ValidatingWebhookConfiguration")
+			return reconcile.Result{}, nil
+		}
+		log.Info("ensuring CA cert on ValidatingWebhookConfiguration")
+		injectCertToWebhook(vwh, artifacts.CertPEM)
+		if err := r.Update(context.Background(), vwh); err != nil {
+			return reconcile.Result{Requeue: true}, nil
```

Why swallow the error? You could return it or at least log it?

---

fixed